### PR TITLE
fix(mergedeep): handle `null` `actor_id` in bypass actor comparison

### LINKED ISSUE #1034: Bypass actors with `actor_id: null` are invisible to ruleset diffs; `actor_id: 1` re-applies forever
For ruleset `bypass_actors`, GitHub's API ignores `actor_id` when `actor_type` is `OrganizationAdmin` (or `DeployKey`) and always returns `actor_id: null` for those entries. `MergeDeep` cannot handle that shape, in either direction:

**Configured as `actor_id: null`** (matching what the API returns): the entry never deploys. `NAME_USERNAME_PROPERTY` picks `actor_id` as the object's identity field because the key exists, `GET_NAME_USERNAME_PROPERTY` returns the null, and `processArrays` skips entries with a falsy id (`if (id) visited[id] = a`). An `OrganizationAdmin` actor present in config but missing from the live ruleset produces no addition, no modification, and a "no changes" plan. The sync silently does nothing.

**Configured as `actor_id: 1`** (what `docs/sample-settings/settings.yml` shows): the entry deploys, but on the next sync the config entry (id `1`) cannot match the live entry (id `null`), so the same change is detected and re-applied on every run, forever. In our org this flapped 15 repos on an hourly schedule until we noticed.

So there is no `actor_id` value for an `OrganizationAdmin` bypass actor that both deploys and converges.

Reproduction with `compareDeep` directly:

```js
const md = new MergeDeep(log, github, [])
const orgAdmin = { actor_id: null, actor_type: 'OrganizationAdmin', bypass_mode: 'pull_request' }
md.compareDeep({ bypass_actors: [] }, { bypass_actors: [orgAdmin] }, {}, {}, {})
// => hasChanges: false  (expected: an addition)
```

### PR DESCRIPTION
**Pull request by tdabasinskas**
Fixes #1034.

GitHub ignores `actor_id` for certain bypass actor types (`OrganizationAdmin`, `DeployKey`) and always returns `null`. Previously, the merge logic would fail to converge because it couldn't properly identify these actors when comparing config to live state.

Changes:
- Add actor_type to IDENTITY_FIELDS for proper actor differentiation
- Skip null-valued identity fields when finding the primary identifier
- Update sample config to use null actor_id for OrganizationAdmin
- Add test coverage for null actor_id scenarios

### CODE REVIEW

File: lib/mergeDeep.js [hunk 1]
```diff
@@ -2,7 +2,8 @@ const mergeBy = require('./mergeArrayBy')
 const DeploymentConfig = require('./deploymentConfig')
 
 const NAME_FIELDS = ['name', 'username', 'actor_id', 'login', 'type', 'key_prefix', 'context']
-const NAME_USERNAME_PROPERTY = item => NAME_FIELDS.find(prop => Object.prototype.hasOwnProperty.call(item, prop))
+const IDENTITY_FIELDS = ['name', 'username', 'actor_id', 'actor_type', 'login', 'type', 'key_prefix', 'context']
+const NAME_USERNAME_PROPERTY = item => IDENTITY_FIELDS.find(prop => Object.prototype.hasOwnProperty.call(item, prop) && item[prop] != null)
 const GET_NAME_USERNAME_PROPERTY = item => { if (NAME_USERNAME_PROPERTY(item)) return item[NAME_USERNAME_PROPERTY(item)] }
```
[the remark this answers is not in the document]
Comment (tdabasinskas) [reply lib/mergeDeep.js:7]: Correct observation, and it's the boundary this PR draws on purpose. Merge-layer matching for null-id actors is unchanged by this change: mergeBy never matched them before either, so layered configs defining the same OrganizationAdmin actor duplicated it in the merged config on stock versions too. The naive fix (adding actor_type to NAME_FIELDS) is worse than the status quo — mergeBy treats a match on any field as identity, so two different actors of the same type (say two teams with different actor_ids) would merge into one during org/suborg/repo layering. Fixing the merge layer properly means making its matcher precedence-aware (first shared field wins, like the diff-identity helpers), which changes matching semantics for every plugin and deserves its own issue and PR rather than riding along here.
